certified-connectors/3E Events/apiProperties.json

```diff
@@ -134,10 +134,340 @@
         "description": "Select the name of the environment."
       },
       "values": [
+        {
+          "name": "aadProd",
+          "uiDefinition": {
+            "displayName": "Production (Elite ID)",
+            "description": "Production (Elite ID)."
+          },
+          "metadata": {
+            "allowSharing": false
+          },
+          "parameters": {
+            "token": {
+              "type": "oauthSetting",
+              "oAuthSettings": {
+                "identityProvider": "oauth2generic",
+                "clientId": "[[DUMMY]]",
+                "clientSecret": "[[DUMMY]]",
+                "redirectMode": "Global",
```

Comment:
Hi @isalzh, 

Could you please change redirectMode from "Global" to "GlobalPerConnector"  as this is the latest update, it is not available yet in the public documentation.

Reply:
After I changed the redirectMode, I got the error from Azure AD:
![image](https://github.com/microsoft/PowerPlatformConnectors/assets/36303904/e4ad5b27-c355-4026-93cb-d679f15cf02d)

Looks like the redirect URL has changed and it broke authentication.
Do you have any suggestions how to fix?

Reply:
In accordance with this: [link](![image](https://github.com/microsoft/PowerPlatformConnectors/assets/36303904/b84b1a4c-68d9-4321-8374-1bf0ea84b34f)) we have added 2 redirect URLs.
Now I can- assume we need to add some more. It is clear what needs to be done for custom connector, by what about preview/prod version of the connector?

Reply:
I just told to management regarding this update. Looks like it would require some unplanned changes into the release we already did. We would like to avoid this for now, as it may require retesting things, production redeployment and so on. We would not want to delay the release of the new version of the connector, it contains some critical changes for further work.

Can we skip this update for now, we will add this as a point for the next release?

Reply:
Hello @isalzh, 
Thank you very much for your response. 

Unfortunately, without this change the certification process will not be processed. 

Along with GlobalPerConnector, please change the redirectUrl as below and ensure that new redirectUrl is added to your client app, also update the redirectUrl here in apiProperties.json with new url.

New redirectURI - https://global.consent.azure-apim.net/redirect/3eevents

Reply:
Hello @vmanoharas 
I have updated both redirectMode and redirectUrl
![image](https://github.com/microsoft/PowerPlatformConnectors/assets/36303904/6d6d55cc-6f17-4e22-a65f-b362fbbf48fe)

Also, added the URL you suggested to the client app:
![image](https://github.com/microsoft/PowerPlatformConnectors/assets/36303904/6c3590e9-e203-4e0a-ab35-74db2cf3c6b6)

Deployed the connector as custom with paconn tool and have the same error:
![image](https://github.com/microsoft/PowerPlatformConnectors/assets/36303904/bd852b49-90f7-47f9-abb7-3b770d61acf3)

Looks like it doesn't honor the redirect URL I setup. How can I test out this to make sure everything is working before I push the changes into the open source repository?

Reply:
Hi @isalzh, 

I apologize for confusion. I've now confirmed with the team, and the recommended action is to remove the "redirecturl" (the entire line) and retest the configuration. This adjustment should give a positive result. However, please ensure to retain "GlobalPerConnector" as it is.

Reply:
Hi @vmanoharas 

Didn't help. Got the same error.
I removed the entire "redirecturl" and leave only GlobalPerConnector:
![image](https://github.com/microsoft/PowerPlatformConnectors/assets/36303904/469be7e8-183d-4bba-bc26-d2495469d75e)

But I noticed that after I uploaded the connector using `paconn update` command, Power Automate auto generated the field with the following value:
![image](https://github.com/microsoft/PowerPlatformConnectors/assets/36303904/b19fef80-0f4e-404b-88ef-2a59862fda70)

I updated `paconn` to the latest version (0.0.21) before.

Reply:
Hi @isalzh, 

Apologize for the inconvenience have caused due to this.  

Can you please refer this link and fix the issue https://learn.microsoft.com/en-us/troubleshoot/azure/active-directory/error-code-aadsts50011-redirect-uri-mismatch.

Reply:
Hello @vmanoharas 
Previously you ask me:

> Along with GlobalPerConnector, please change the redirectUrl as below and **ensure that new redirectUrl is added to your client app**, also update the redirectUrl here in apiProperties.json with new url.
New redirectURI - **https://global.consent.azure-apim.net/redirect/3eevents**

I added, but I still can see that connector is trying to request access token using **another**, self generated redirectURL.
It's ok I can add the new one as well for non-prod application. But I have to be 100% sure that after certification the connector will send requests with this URL ****https://global.consent.azure-apim.net/redirect/3eevents**** since I cannot simulate this in non-prod. Can you guarantee this?

Aren't these URLs expected to be different for preview and prod Power Automate?

I apologize for asking all these questions to you, but it is not completely clear how the new things work, without documentation.

Reply:
Hi @isalzh, 

Please accept my sincere apology for asking you many questions and changes.  Please revert the changes and. Our team will guide you through the changes.  As this is needed during certification process.

Reply:
Hi @vmanoharas

These changes are very important to us and our customers and we cannot delay them. Could you pay special attention to this?

Reply:
Sure @isalzh, I am approving this.